cmd/hiffy/src/lib.rs

```diff
@@ -401,13 +403,19 @@ fn hiffy(context: &mut humility::ExecutionContext) -> Result<()> {
     let subargs = HiffyArgs::try_parse_from(subargs)?;
 
     if subargs.list {
-        hiffy_list(hubris, subargs.verbose)?;
+        hiffy_list(hubris, subargs.filter)?;
         return Ok(());
+    } else if !subargs.filter.is_empty() {
+        bail!("filters can only be provided with --list");
     }
 
     let mut context = HiffyContext::new(hubris, core, subargs.timeout)?;
 
     if let Some(call) = subargs.call {
+        if core.is_dump() {
```

Comment:
This check should go above `HiffyContext::new`; otherwise, HIF version mismatches cause it to fail first with a misleading error message:
```
➜  humility git:(test) ✗ cargo run -- -d tests/cmd/cores/hubris.core.kiowa.27 hiffy -c UserLeds.led_on
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
     Running `target/debug/humility -d tests/cmd/cores/hubris.core.kiowa.27 hiffy -c UserLeds.led_on`
humility: attached to dump
humility hiffy failed: HIF version mismatch: target has 0.2; ours is 0.3
```

Reply:
That's just a consequence of trying to run it on a very old dump. If you run it on a dump in which the HIF version doesn't mismatch, you'll get the expected error:

```
$ ./target/release/humility -d ./tests/cmd/cores/hubris.core.sprot_status hiffy -c Validate.validate_i2c
humility: attached to dump
humility hiffy failed: can't make HIF calls on a dump
```

And leaving it there allows for HIF functions to also be listed out of a dump, which has utility:

```
$ ./target/release/humility -d ./tests/cmd/cores/hubris.core.sprot_status hiffy -L
humility: attached to dump
 ID FUNCTION                       #ARGS
  0 Sleep                          1
  1 Send                           4
  2 SendLeaseRead                  5
  3 SendLeaseWrite                 5
  4 I2cRead                        7
  5 I2cWrite                       8
  6 I2cBulkWrite                   8
  7 GpioInput                      1
  8 GpioToggle                     2
  9 GpioSet                        2
 10 GpioReset                      2
 11 GpioConfigure                  7
 12 SpiRead                        4
 13 SpiWrite                       3
 14 QspiReadId                     0
 15 QspiReadStatus                 0
 16 QspiBulkErase                  0
 17 QspiPageProgram                3
 18 QspiRead                       2
 19 QspiSectorErase                1
 20 QspiVerify                     3
 21 QspiHash                       2
 22 HashDigest                     1
 23 HashInit                       0
 24 HashUpdate                     1
 25 HashFinalize                   0
 26 StartUpdate                    0
 27 WriteBlock                     2
 28 FinishUpdate                   0
 29 BlockSize                      0
 30 SpRotSendRecv                  2
 31 SpRotSendRecvRetries           3
 32 SpRotPulse                     1
 33 SpRotSink                      2
 34 SpRotStatus                    0
 35 SpRotStartUpdate               0
 36 SpRotWriteBlock                2
 37 SpRotFinishUpdate              0
 38 SpRotBlockSize                 0
```

Reply:
Yeah, I understand that; I'm thinking about the case where you've specified `--call`,  are running on a dump, **and** have a HIF mismatch.  In that case, I think "can't make HIF calls on a dump" message should be prioritized over the HIF mismatch error.

e.g.
```diff
@@ -409,13 +427,12 @@ fn hiffy(context: &mut humility::ExecutionContext) -> Result<()> {
         bail!("filters can only be provided with --list");
     }

+    if subargs.call.is_some() && core.is_dump() {
+        bail!("can't make HIF calls on a dump");
+    }
     let mut context = HiffyContext::new(hubris, core, subargs.timeout)?;

     if let Some(call) = subargs.call {
-        if core.is_dump() {
-            bail!("can't make HIF calls on a dump");
-        }
-
         let func: Vec<&str> = call.split('.').collect();

         if func.len() != 2 {
```

This should still allow you to list functions on a dump with a HIF mismatch.